### PR TITLE
DASD-8885 split outer

### PR DESCRIPTION
**Pull request by adam230594**
- Use new AKS deployment agents
- Pipeline per outer API
- ASP deployment now deploys via shared-infrastructure

### CODE REVIEW

File: deployments/apprentice-commitments.yml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  displayName: Deploy to AT
+  variables:
+  - group: DevTest Management Resources
+  - group: AT DevTest Shared Resources
+  - group: AT das-apim-endpoints
+  jobs:
+  - template: ../pipeline-templates/job/arm-deploy.yml
+    parameters:
+      ServiceConnection: SFA-DAS-DevTest-ARM
+      AppRoleAssignmentsServiceConnection: das-app-role-assignments-CDS
```
Comment (jack-cloud-platform) [inline deployments/apprentice-commitments.yml:45]: Should the task that uses this service connection only be in this job template, and not deploy-api.yml as well

File: deployments/apprentice-commitments.yml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+stages:
+- stage: Deploy_AT
+  dependsOn: []  
+  displayName: Deploy to AT
+  variables:
+  - group: DevTest Management Resources
+  - group: AT DevTest Shared Resources
+  - group: AT das-apim-endpoints
+  jobs:
+  - template: ../pipeline-templates/job/arm-deploy.yml
```
Comment (jack-cloud-platform) [inline deployments/apprentice-commitments.yml:42]: Could we pass another parameter for schema file to process? So only the single Outer API's file is processed?

Comment (adam230594) [reply deployments/apprentice-commitments.yml:42]: Great suggestion. Will get this applied 👍

File: deployments/apprentice-commitments.yml (hunk 1 above)
Comment (adam230594) [reply deployments/apprentice-commitments.yml:45]: The app role assignments task is ran in both arm-deploy.yml as well as deploy-api.yml

arm-deploy applies the role assignment for APIM -> outer API's and deploy-api.yaml applies the role assignment for outer API's -> inner API's

File: pipeline-templates/job/arm-deploy.yml [hunk 3]
```diff
@@ -2,12 +2,12 @@ parameters:
   ServiceConnection: ''
   AppRoleAssignmentsServiceConnection: ''
   Environment: ''
+  SchemaFilePath: '*.schema.json'
```
Comment (jack-cloud-platform) [inline pipeline-templates/job/arm-deploy.yml:5]: Could this have an empty default value?

**Review by jack-cloud-platform — APPROVED** [review]

**Pull request merged**